tests/src/unit-regression2.cpp

```diff
@@ -1094,6 +1110,14 @@ TEST_CASE("regression tests 2")
         CHECK(j.type_name() == "invalid");
     }
 #endif
+
+#ifdef JSON_HAS_CPP_17
+    SECTION("issue #4804: from_cbor incompatible with std::vector<std::byte> as binary_t")
+    {
+        const std::vector<char> data;
+        const auto decoded = json_4804::from_cbor(data);
```

Comment:
Is this test-case even supposed to run successfully? I'm aware that I posted this test-case when raising the issue, but I was expecting it to compile, not necessarily to actually run successfully. This attempts to decode the empty byte-string as CBOR. While this *may* be valid CBOR (though [CBOR playground](https://cbor.me/) rejects it), it's unclear to me how that would be represented within the JSON object model. In particular, the CBOR representation of the empty object is not the same as the empty byte-string. So perhaps `"\xA0"` (the correct CBOR representation of the empty map/object) would be a better test-case.

Reply:
Good catch. Yes, I indeed just copied your example and made it compile. Very strange to see that some compilers seem to allow parsing from an empty vector.